### PR TITLE
Refactoring VolumeId Tests to use gomock; Adding additional tests to assert the calls to functions

### LINKED ISSUE #31: Refactor AWS EC2 unit tests to use `gomock`
Currently many of the unit tests for the EC2 related methods in the `aws` package use a home-built mock object for the AWS client. To get more consistency and feature-availability for addressing test gaps, these methods should be converted over to use `gomock` based methods similar to those in the DAX related methods (as an example). The mocked objects are already present; we just need to convert the tests themselves to using them.

As part of this, testing gaps found in #30 should be addressed as well. This includes making it so that the volume with the non-matching Device ID has throughput, type, and IOPS values that do _not_ match the asserted ones. This way we can be sure that the methods are selecting only the specified device.

### PR DESCRIPTION
**Pull request by sunkaraamrutha**
### SUMMARY
* Currently many of the unit tests for the EC2 related methods in the aws package use a home-built mock object for the AWS client. To get more consistency and feature-availability for addressing test gaps, these methods should be converted over to use gomock based methods similar to those in the DAX related methods (as an example). The mocked objects are already present; we just need to convert the tests themselves to using them.
* As part of this, testing gaps found in https://github.com/HBOCodeLabs/infratest/pull/30 should be addressed as well. This includes making it so that the volume with the non-matching Device ID has throughput, type, and IOPS values that do not match the asserted ones. This way we can be sure that the methods are selecting only the specified device.

This fixes #31 .
### DEVELOPER IMPACT
* NA
* If there are new shared libraries required that are not listed in the list in the project README, please explain briefly why they are needed.
### Contribution Checklist
* [x] All changes have been reflected in comparable unit test changes or additions.
* [x] Any interactions with third party clients are done via interface types rather than direct interactions.
* [x] All new functions follow the required naming standard.
* [x] All new functions follow the required signature standards.

### CODE REVIEW

File: pkg/aws/ec2_test.go [hunk 1]
```diff
@@ -928,6 +952,7 @@ func TestAssertEC2VolumeType_MatchWithThroughputMultipleDevices(t *testing.T) {
 
 func TestAssertEC2VolumeType_MatchWithIOPSMultipleDevices(t *testing.T) {
 	// Setup
+	// t.Parallel()
```
Comment (yardbirdsax) [inline pkg/aws/ec2_test.go:955]: **Suggestion:** Would you please uncomment this, unless there's a reason this test can't be run in parallel, in which case would you please remove the commented code?

**Review by yardbirdsax — CHANGES_REQUESTED** [review]
Thanks for the PR @sunkaraamrutha ! One small tweak and this should be good. Thanks for paying down some tech debt.

File: pkg/aws/ec2_test.go (hunk 1 above)
Comment (sunkaraamrutha) [reply pkg/aws/ec2_test.go:955]: addressed

**Review by yardbirdsax — APPROVED** [review]
Approved, LGTM

**Pull request merged**